### PR TITLE
GRASS GIS 8.0 changes

### PR DESCRIPTION
**Pull request by neteler**
This PR ~~is meant for testing (do not merge into master yet!) as it~~ contains needed changes for a GRASS GIS 8.0 ~~.0alpha release~~.
~~It may be used for local testing by applying this PR to the latest GRASS GIS-core in the master branch.~~

_Update 28 May 2021:_
This PR may become the basis for the changes in the master branch to GRASS GIS 8.0. 
A  8.0.0alpha release branch may be forked after merge of this PR.

Please verify thoroughly.

(note that some CI tests will fail as they expect `grass79` and not `grass80` as the name of startup script; in addition, there is no `grass8` branch yet in addons)

To locally apply and try it:

1. checkout GRASS GIS master from GitHub, `cd` into it
2. if you already have it, `make distclean`
2. `wget https://github.com/OSGeo/grass/pull/1597.diff`
3. `patch -p1 < 1597.diff`
4. `configure ... ; make`
5. start it from `bin.x86_64-pc-linux-gnu/grass80` (the name of architecture in the path may differ)

### CODE REVIEW

File: NEWS
Comment (hellik) [inline NEWS:4]: Grass7 => Grass8?

File: REQUIREMENTS.html
Comment (hellik) [inline REQUIREMENTS.html:58]: what is the minimum Python 3.x? really 3.5?

File: db/db.login/db.login.html
Comment (hellik) [inline db/db.login/db.login.html:19]: wouldn't it be

`C:\Users\youruser\AppData\Roaming\GRASS7`

?

Comment (hellik) [reply db/db.login/db.login.html:19]: in this case `%APPDATA%\Roaming\GRASS8`

File: doc/grass_database.html
Comment (hellik) [inline doc/grass_database.html:225]: that's starting a text session in *nix. should we add an example how to do it in windows?

File: NEWS
Comment (neteler) [reply NEWS]: Good catch, fixed in 8078cc676df18473fc3aceee6b9f7a78822b46e9

File: REQUIREMENTS.html
Comment (neteler) [reply REQUIREMENTS.html:58]: That's still to be defined... 

See RFC 8: Language Standards Support: Python (Draft), https://trac.osgeo.org/grass/wiki/RFC/8_LanguageStandardsSupport

File: db/db.login/db.login.html
Comment (neteler) [reply db/db.login/db.login.html]: ok, fixed in 9ee56ed3c

File: doc/grass_database.html
Comment (neteler) [reply doc/grass_database.html:225]: Yes, please donate an example (here is fine)

File: python/grass/script/setup.py
Comment (hellik) [inline python/grass/script/setup.py:32]: do we provide still a 32 bit standalone installer C:\Program Files (x86)? @landam

**Review by hellik — CHANGES_REQUESTED** [review]
left mostly windows related comments

File: doc/grass_database.html
Comment (hellik) [reply doc/grass_database.html:225]: grass80 --text D:\grassdata\nc_spm_08_grass7\user1

File: doc/grass_database.html
Comment (hellik) [reply doc/grass_database.html:233]: grass80 -c EPSG:31287 D:\grassdata\mylocation_epsg31287

File: gui/wxpython/docs/wxGUI.html
Comment (neteler) [reply gui/wxpython/docs/wxGUI.html]: fixed in 7704a7a8f

File: lib/init/variables.html
Comment (neteler) [reply lib/init/variables.html]: fixed in 7704a7a8f

File: doc/grass_database.html
Comment (neteler) [reply doc/grass_database.html:238]: done (modified) in d83447807

File: doc/grass_database.html
Comment (neteler) [reply doc/grass_database.html]: done in d83447807

File: gui/wxpython/docs/wxGUI.toolboxes.html
Comment (neteler) [reply gui/wxpython/docs/wxGUI.toolboxes.html:21]: yes, pls donate an example

**Comment by wenzeslaus** [comment]
> To locally apply and try it:
>  ...
>     3. `wget https://github.com/OSGeo/grass/pull/1597.diff`
>     4. `patch -p1 < 1597.diff`

Or `gh pr checkout 1597` following the _Open with_ button right from the PR title. (Or manually, add the neteler fork repo as another remote to your local repo and get the grass_gis_800_alpha from there.)

File: doc/grass_database.html
Comment (wenzeslaus) [inline doc/grass_database.html:230]: Do you think that someone who has an issue figuring out the need for a Windows path from the unix example (as before) will parse through the `#` comments in the example? Either delete the Windows example or make it a separate paragraph.

File: INSTALL
Comment (wenzeslaus) [inline INSTALL:154]: What about dropping the 80 part at least from the documentation? On a reasonable system, it should install as `grass` anyway. I also noticed I usually write `ls` and not `ls830`, `gdalinfo`, not `gdalinfo32`... Seems like less work for the releases, too.

File: INSTALL
Comment (wenzeslaus) [inline INSTALL:192]: This should definitively say `grass-source-code` without the number.

File: REQUIREMENTS.html
Comment (wenzeslaus) [reply REQUIREMENTS.html:58]: Since, as the RFC says, _the oldest still-alive Python version is 3.6, to be retired in December 2021_, I don't see any reason to have anything lower than 3.6 here.

File: db/db.login/db.login.html
Comment (wenzeslaus) [inline db/db.login/db.login.html:19]: Doesn't `/` just work on Windows [these days](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.altdirectoryseparatorchar?view=net-5.0), @hellik?

**Comment by wenzeslaus** [comment]
> This PR is meant for testing (do not merge into master!) as it contains needed changes for a GRASS GIS 8.0.0alpha release.

I don't understand where this will go. If it will be released, it should go to the main repo. A branch would make sense to me.

**Comment by wenzeslaus** [comment]
> (note that some CI tests will fail as they expect `grass79` and not `grass80` as the name of startup script)

Can we install just `grass` to avoid this?

File: doc/grass_database.html
Comment (veroandreo) [reply doc/grass_database.html:230]: What do you mean? In my experience, students want to see it written, they do not want to figure it out themselves...

File: lib/init/variables.html
Comment (veroandreo) [reply lib/init/variables.html:56]: would you please provide some, @hellik ?

File: python/grass/script/core.py
Comment (veroandreo) [inline python/grass/script/core.py:350]:
```suggestion
                        " deprecated in GRASS GIS 7.0 and has been removed"
```

File: db/db.login/db.login.html
Comment (HuidaeCho) [reply db/db.login/db.login.html:19]: / works on Windows (https://answers.microsoft.com/en-us/windows/forum/windows_7-files/when-did-windows-start-accepting-forward-slash-as/aa033741-0fc6-43e3-ab6e-165020794126), but I don't think it's a good idea to use / in general (https://bytes.com/topic/python/answers/23123-when-did-windows-start-accepting-forward-slash-path-separator), not in this specific case.
```dos
cd /users # works fine
mkdir \dev
cd /dev # doesn't work because /d is an option
```

File: INSTALL
Comment (neteler) [reply INSTALL:154]: Generally fine for me but then we may also need to actually generate a `grass` script at compile time.

File: INSTALL
Comment (neteler) [reply INSTALL]: fixed in 5bd452e4e

File: REQUIREMENTS.html
Comment (neteler) [reply REQUIREMENTS.html]: Updated to Python 3.6 in dcfb6c548

File: python/grass/script/core.py
Comment (neteler) [reply python/grass/script/core.py]: fixed here and elsewhere in b7c892155

File: INSTALL
Comment (neteler) [reply INSTALL:154]: Yes, fine for me.

> > This PR is meant for testing (do not merge into master!) as it contains needed changes for a GRASS GIS 8.0.0alpha release.
> 
> I don't understand where this will go.

As I understand there are still different opinions (https://lists.osgeo.org/pipermail/grass-dev/2021-May/095136.html and related) on when to create a branch.

>  If it will be released, it should go to the main repo. A branch would make sense to me.

We could asap, as I don't see much point in keeping "7.9" forever:
- merge this PR into the main branch (changing VERSION from "8.0.0alpha" to "8.0.dev")
- then create the "releasebranch_8_0" branch off the main branch (changing VERSION therein to "8.0.0alpha")
- cherry-pick future relevant changes from main branch to "releasebranch_8_0" as we have done with other release branches

**Comment by neteler** [comment]
> > (note that some CI tests will fail as they expect `grass79` and not `grass80` as the name of startup script)
> 
> Can we install just `grass` to avoid this?

Would be practical, however, I routinely use the stable and the dev versions in parallel. But that could be solved by a link...

File: doc/grass_database.html
Comment (wenzeslaus) [reply doc/grass_database.html:230]: My question is how large is the overlap between the groups: Group 1 can replace unix path by Windows path easily. Group 2 is not distracted or confused by the `#` comments. I'm guessing pretty small. In other words, if the reader is familiar with unix/Bash syntax, the Windows piece is not needed. And if a unix path is a stumbling block, so is a Bash comment.

**Comment by wenzeslaus** [comment]
> > > (note that some CI tests will fail as they expect `grass79` and not `grass80` as the name of startup script)
> > 
> > 
> > Can we install just `grass` to avoid this?
> 
> Would be practical, however, I routinely use the stable and the dev versions in parallel. But that could be solved by a link...

Let's not make it hard everywhere and for the users because developers need to have multiple versions.

**Comment by wenzeslaus** [comment]
> We could asap, as I don't see much point in keeping "7.9" forever:
>     * merge this PR into the main branch (changing VERSION from "8.0.0alpha" to "8.0.dev")
>     * then create the "releasebranch_8_0" branch off the main branch (changing VERSION therein to "8.0.0alpha")
>     * cherry-pick future relevant changes from main branch to "releasebranch_8_0" as we have done with other release branches

Or you could do the 8.0dev part, but branch out only to create the alpha and then abandon that branch and create the 8.0 release branch later avoiding any backports between the alpha and the more mature release branching.

In any case, the "odd versions are dev and never released" system seems to break here, but after creating 8.0 release branch, master branch would need to be 8.1.

**Comment by neteler** [comment]
> Let's not make it hard everywhere and for the users because developers need to have multiple versions.

Ok, what do you suggest? I didn't get it.

**Comment by wenzeslaus** [comment]
> > Let's not make it hard everywhere and for the users because developers need to have multiple versions.
> 
> Ok, what do you suggest? I didn't get it.

That installing `grass` is the expected behavior and it should be the default.

File: doc/grass_database.html
Comment (hellik) [reply doc/grass_database.html:230]: in my experience, many manual examples are too *nix biased and exclude _average_ windows user by not giving literal working windows examples. I agree here with @veroandreo.

**Comment by neteler** [comment]
> > > Let's not make it hard everywhere and for the users because developers need to have multiple versions.
> That installing `grass` is the expected behavior and it should be the default.

Ok, I have changed it (82e8429e8a6c20a88275720447147b6c66a74dfe), it naturally implied other places to be changed.

File: lib/init/variables.html
Comment (hellik) [reply lib/init/variables.html:56]: it's again in  `%APPDATA%\Roaming\GRASS8\` :

`%APPDATA%\Roaming\GRASS8\rc`

**Comment by neteler** [comment]
> > (note that some CI tests will fail as they expect `grass79` and not `grass80` as the name of startup script)
> 
> Can we install just `grass` to avoid this?

Done (82e8429) - now it would be great to have the CI accept `grass` as well...

File: lib/init/variables.html
Comment (neteler) [reply lib/init/variables.html]: added in efe90a9f7

File: db/db.login/db.login.html
Comment (hellik) [reply db/db.login/db.login.html:19]: > cd /data  # doesn't work because /d is an option

AFAIK it's 

`cd data`

`cd \data` changes here to an existing directory in the toplevel d: , altough I've started e.g. in D:\temp\tm18> where the data subfolder exists.

```
D:\temp\tm18>cd /data
Das System kann den angegebenen Pfad nicht finden.
```

Comment (hellik) [reply db/db.login/db.login.html:19]: > 
> 
> Doesn't `/` just work on Windows [these days](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.altdirectoryseparatorchar?view=net-5.0), @hellik?

the interesting thing is - and almost nobody knows it ;-)  - it worked since the beginning of MS windows

```
D:\>cd D:\temp\tm18\data
D:\temp\tm18\data>
```

```
D:\>cd D:/temp/tm18/data
D:\temp\tm18\data>
```

though the MS windows standard, everywhere promoted, is:

`D:\temp\tm18\data`

and you can't be sure that it works in all circumstances of myriads of different varieties of batch files, python scripts, R scripts in windows .....

File: INSTALL
Comment (landam) [inline INSTALL:217]: What about simply `gmake`?

Comment (neteler) [reply INSTALL]: Deal: 98388f565

File: db/db.login/db.login.html
Comment (wenzeslaus) [reply db/db.login/db.login.html:19]: I wonder if there are any circumstances where it is still an issue except for that `/d` issue. Python scripts should never have explicit slash or backslash in them. The best practices for R I have seen argued for forward slashes on all platforms (saying to make them work on macOS and Linux, so they actually seemed written from Windows point of view).

**Review by wenzeslaus — COMMENTED** [review]
The `grass79` -> `grass` seems like the best part of this PR!

**Comment by neteler** [comment]
> > Done ([82e8429](https://github.com/OSGeo/grass/commit/82e8429e8a6c20a88275720447147b6c66a74dfe)) - now it would be great to have the CI accept `grass` as well...
> 
> You can just change it together with the rest of the files. Your find and replace should work.

Ok, done in d885e67d2

**Comment by neteler** [comment]
> > You can just change it together with the rest of the files. Your find and replace should work.
> 
> Ok, done in [d885e67](https://github.com/OSGeo/grass/commit/d885e67d2206cb9727d8897ba44990b5400c487b)

Something is still missing concerning the CI (I wonder where...):

Ubuntu log:

```
mkdir -p -m 755 /home/runner/install//grass80
mkdir -p -m 755 /home/runner/install//bin
tar cBCf /home/runner/work/grass/grass/dist.x86_64-pc-linux-gnu - . | tar xBCf /home/runner/install//grass80 - 2>/dev/null
rm /home/runner/install//grass80/grass80.tmp
rm: cannot remove '/home/runner/install//grass80/grass80.tmp': No such file or directory
```

**Comment by wenzeslaus** [comment]
It looks like a problem with the `make install` step. Seem to be coming from `include/Make/Install.make:101: -rm $(DESTDIR)$(INST_DIR)/$(GRASS_NAME).tmp`. Does it work locally? (Can't investigate the GRASS_NAME piece now.)

File: doc/grass_database.html
Comment (wenzeslaus) [reply doc/grass_database.html]: > ...many manual examples are too *nix biased...

I think what would help is saying that it is a Bash or whatever example so that it would be clear that commands or syntax may be different (on Windows) and that it is not "standard GRASS" but rather "standard Bash/unix/...".

Otherwise, I think we need Python examples in manual pages more than Windows examples anyway.

> ...exclude average windows user...

Similarly to the "dropping terminal" PR, I continue being puzzled about who are these _average Windows users_ attached to their command lines. Isn't that a stereotype about Linux users, not Windows ones? :-)

**Comment by neteler** [comment]
> It looks like a problem with the `make install` step. 

Found it: fcb524dfc

File: INSTALL
Comment (ninsbl) [reply INSTALL]: > Generally fine for me but then we may also need to actually generate a `grass` script at compile time.

A version independend startup script is nice if this does not interfere with installing different GRASS versions in parallel.
So we might need to keep `grass78` and `grass80` anyway? Just to make sure different version in parallel are considered...

**Comment by wenzeslaus** [comment]
It seems like you can re-conceptualize this PR to be a 7-to-8 transition, merge it to master branch, and do the alpha release changes elsewhere. Thoughts?

**Comment by neteler** [comment]
Getting pretty close to "CI passing", yet this one I cannot identify in the code/workflow:

```
Run .github/workflows/test_simple.bat 'C:\OSGeo4W64\opt\grass\grass.bat'

D:\a\grass\grass>set grass=C:\OSGeo4W64\opt\grass\grass.bat 

D:\a\grass\grass>call C:\OSGeo4W64\opt\grass\grass.bat --tmp-location EPSG:4326 --exec g.region res=0.1 -p 
(null): can't open file 'C:\OSGEO4~1\opt\grass\etc\grass80.py': [Errno 2] No such file or directory
Press any key to continue . . . 
Error: Process completed with exit code 1.
```

**Comment by nilason** [comment]
> It seems like you can re-conceptualize this PR to be a 7-to-8 transition, merge it to master branch, and do the alpha release changes elsewhere. Thoughts?

A lot of the changes, certainly the ones related to `grass79->grass` and `grass8` in docs need to be merged into master before any branching. I would like to suggest to merge this PR (when approved) and give it some days/a week for testing the grassXX->grass (👍 BTW) changes in particular. Then tag a 8.0.0.alpha1 on master for release. It's not unlikely there will pop up issues quite immediately after testing in a wider group, these issues can hopefully quickly be addressed and applied directly to master. When these immediate issues are ironed out (I'm thinking of a time frame of about 2-4 weeks) a 8.0 branch may be made, perhaps starting with a beta version or - even better if feasable - a release candidate.

**Comment by nilason** [comment]
> Getting pretty close to "CI passing", yet this one I cannot identify in the code/workflow:
> 
> ```
> Run .github/workflows/test_simple.bat 'C:\OSGeo4W64\opt\grass\grass.bat'
> 
> D:\a\grass\grass>set grass=C:\OSGeo4W64\opt\grass\grass.bat 
> 
> D:\a\grass\grass>call C:\OSGeo4W64\opt\grass\grass.bat --tmp-location EPSG:4326 --exec g.region res=0.1 -p 
> (null): can't open file 'C:\OSGEO4~1\opt\grass\etc\grass80.py': [Errno 2] No such file or directory
> Press any key to continue . . . 
> Error: Process completed with exit code 1.
> ```

Could it be:
https://github.com/OSGeo/grass/blob/2048e101650b587d641dbe0d7ec561f84ec1194b/mswindows/osgeo4w/grass.bat.tmpl#L18

**Comment by neteler** [comment]
> > D:\a\grass\grass>call C:\OSGeo4W64\opt\grass\grass.bat --tmp-location EPSG:4326 --exec g.region res=0.1 -p 
> > (null): can't open file 'C:\OSGEO4~1\opt\grass\etc\grass80.py': [Errno 2] No such file or directory
> Could it be:
> 
> https://github.com/OSGeo/grass/blob/2048e101650b587d641dbe0d7ec561f84ec1194b/mswindows/osgeo4w/grass.bat.tmpl#L18

Ah, thanks. Changed in ec8e141a0

**Comment by nilason** [comment]
> > > D:\a\grass\grass>call C:\OSGeo4W64\opt\grass\grass.bat --tmp-location EPSG:4326 --exec g.region res=0.1 -p
> > > (null): can't open file 'C:\OSGEO4~1\opt\grass\etc\grass80.py': [Errno 2] No such file or directory
> > > Could it be:
> > 
> > 
> > https://github.com/OSGeo/grass/blob/2048e101650b587d641dbe0d7ec561f84ec1194b/mswindows/osgeo4w/grass.bat.tmpl#L18
> 
> Ah, thanks. Changed in [ec8e141](https://github.com/OSGeo/grass/commit/ec8e141a04d6130f99a64fc4582a58fcfaef9f6e)

I did notice `POSTFIX` is used in several places and files in similar way. Remain to be see CIs turnout, but may be tested with `export POSTFIX=""`, or evaluate case by case.

**Comment by nilason** [comment]
I see it fails again:
```
D:\a\grass\grass>call C:\OSGeo4W64\opt\grass\grass.bat --tmp-location EPSG:4326 --exec g.region res=0.1 -p 
Starting GRASS GIS...
Traceback (most recent call last):
  File "C:\OSGEO4~1\opt\grass\etc\grass.py", line 2721, in <module>
    main()
  File "C:\OSGEO4~1\opt\grass\etc\grass.py", line 2595, in main
    tmpdir=tmpdir,
  File "C:\OSGEO4~1\opt\grass\etc\grass.py", line 1035, in set_mapset
    from grass.grassdb.checks import (
ModuleNotFoundError: No module named 'grass.grassdb'; 'grass' is not a package
Press any key to continue . . . 
Error: Process completed with exit code 1.
```
my guess it's related to `POSTFIX` (looking for the module at a path including non-existing `grass80`) as I noted above. Maybe you could try set it to an empty string.

**Comment by neteler** [comment]
> I see it fails again:
...
> my guess it's related to `POSTFIX` (looking for the module at a path including non-existing `grass80`) as I noted above. Maybe you could try set it to an empty string.

Ok, I try that now: 1fe7743ab

File: mswindows/osgeo4w/package.sh
Comment (hellik) [inline mswindows/osgeo4w/package.sh:114]: we have this logic with ${MAJOR}${MINOR} that we're able to build/compile e.g. 7.6, 7.8, 7.9, 8.0 etc in the same OSGeo4W build environment side by side; IIRC also used later for compiling the winGRASS addons for different versions.

so if we decide not to use ${MAJOR}${MINOR}, then maybe OSGeo4W build environment has to be re-organized.

@landam what are your suggestions?

**Comment by wenzeslaus** [comment]
>  `ModuleNotFoundError: No module named 'grass.grassdb'; 'grass' is not a package`

My guess: The directory with grass.py is on Python (import) path (`sys.path`). Likely because it is the current directory. Python picks grass.py instead of the grass package. Startup ends in flames.

**Comment by neteler** [comment]
> It seems like you can re-conceptualize this PR to be a 7-to-8 transition, merge it to master branch, and do the alpha release changes elsewhere. Thoughts?

Meanwhile I also believe that this PR is the preparation of the master branch for G8. So.
- a 8.0.0alpha branch shall just be a "throw away/ignore later" branch, created after merging of this PR
- the real release_branch_8_0 can then be created at a later stage, after gaining some experience with the "throw away" 8.0.0alpha branch.

**Comment by neteler** [comment]
> > `ModuleNotFoundError: No module named 'grass.grassdb'; 'grass' is not a package`
> 
> My guess: The directory with grass.py is on Python (import) path (`sys.path`). _Likely because it is the current directory._ Python picks grass.py instead of the grass package. Startup ends in flames.

I see. Shall I revert the parts concerning Windows at this point? I happily accept suggestions how to deal with this.

**Comment by neteler** [comment]
> > `ModuleNotFoundError: No module named 'grass.grassdb'; 'grass' is not a package`
> 
> My guess: The directory with grass.py is on Python (import) path (`sys.path`). _Likely because it is the current directory._ Python picks grass.py instead of the grass package. Startup ends in flames.

Does anyone have an idea how to deal with this collision? It's the remaining showstopper to merge this PR.

**Comment by neteler** [comment]
> Shall I revert the parts concerning Windows at this point? 

Ok, I have reverted all Windows related changes for now.
The only Windows related change is "grass79" -> "grass80".

**Comment by landam** [comment]
> The only Windows related change is "grass79" -> "grass80" (ignoring any documentation related issues which may be addressed later/in a separate PR).

The Windows issue could be solved in separated PR. Lets move on with important PR.

**Comment by neteler** [comment]
> The Windows issues can be solved in separated PR. Lets move on with this important PR.

With the CI hack in ea31d27 also OSGeo4W builds again. The CI only fails because of the not (yet) existing `grass8` branch in the addons repo.

I suggest to merge this PR.

**Review by nilason — APPROVED** [review]
Shouldn't the following lines:

https://github.com/OSGeo/grass/blob/e5379bbd7e534071eae392bf416865fdbf109f01/configure.in#L244

https://github.com/OSGeo/grass/blob/e5379bbd7e534071eae392bf416865fdbf109f01/configure#L1476

be changed to:
`INSTDIR='${prefix}'"/grass"`
?

Or is this needed for Windows build?

Else, fine for me.

**Comment by neteler** [comment]
> Shouldn't the following lines:
...
> be changed to:
> `INSTDIR='${prefix}'"/grass"`
> ?
> 
> Or is this needed for Windows build?

I'd keep it for now and move on.

**Comment by nilason** [comment]
> > Shouldn't the following lines:
> > ...
> > be changed to:
> > `INSTDIR='${prefix}'"/grass"`
> > ?
> > Or is this needed for Windows build?
> 
> I'd keep it for now and move on.

I needed that change on Mac at least, compilation (or was it install?) failed without it. But that can be fixed with other issues which will likely pop up after merge.

**Comment by nilason** [comment]
> > > Shouldn't the following lines:
> > > ...
> > > be changed to:
> > > `INSTDIR='${prefix}'"/grass"`
> > > ?
> > > Or is this needed for Windows build?
> > 
> > 
> > I'd keep it for now and move on.
> 
> I needed that change on Mac at least, compilation (or was it install?) failed without it. But that can be fixed with other issues which will likely pop up after merge.

I now tested again, working fine. Please ignore my previous comment.

**Comment by HuidaeCho** [comment]
Dropping the version from the startup script broke my cross-compilation cron jobs. Have we decided to not use `${MAJOR}${MINOR}`? What was wrong with it?

**Comment by neteler** [comment]
> Dropping the version from the startup script broke my cross-compilation cron jobs. Have we decided to not use `${MAJOR}${MINOR}`? What was wrong with it?

It has been kept for Windows, and dropped elsewhere. Maybe @wenzeslaus and @nilason can comment here?

**Comment by HuidaeCho** [comment]
> It has been kept for Windows, and dropped elsewhere.

And why is this inconsistency? Quoting @ninsbl

> A version independend startup script is nice if this does not interfere with installing different GRASS versions in parallel. So we might need to keep grass78 and grass80 anyway? Just to make sure different version in parallel are considered...

**Comment by wenzeslaus** [comment]
> Have we decided to not use `${MAJOR}${MINOR}`? What was wrong with it?

Quoting my initial comment above:
 
> What about dropping the 80 part at least from the documentation? On a reasonable system, it should install as `grass` anyway. I also noticed I usually write `ls` and not `ls830`, `gdalinfo`, not `gdalinfo32`... Seems like less work for the releases, too.

In other words, nobody types `vim81` or `ogr2ogr321`. The only exception I can think of is Python where you may have `python`, `python3`, `python3.8`, `python2`, `python2.7`, but that's an extreme case. Moreover, it is a packaging issue more than a basic install issue.

Additionally, the current version in the name is misleading and would require a change anyway. Q: What the `80` in `grass80` means? A: Perhaps that it was not updated since the 80s.

> > It has been kept for Windows, and dropped elsewhere.
> 
> And why is this inconsistency?

Some Python import issue needs to be resolved to fix the Windows install, but that's not a reason to compromise Linux or macOS. The expectations for executables differ anyway between these platforms.

> > A version independend startup script is nice if this does not interfere with installing different GRASS versions in parallel. So we might need to keep grass78 and grass80 anyway? Just to make sure different version in parallel are considered...

Quoting earlier comments:

> > > > (note that some CI tests will fail as they expect `grass79` and not `grass80` as the name of startup script)
> > > 
> > > Can we install just `grass` to avoid this?
> > 
> > Would be practical, however, I routinely use the stable and the dev versions in parallel. But that could be solved by a link...
> 
> Let's not make it hard everywhere and for the users because developers need to have multiple versions.

In other words, the default result you get should be `grass` because that's expected. If someone is managing different versions, they likely have the skills to deal with that accordingly. There is many ways how parallel versions of one software are managed (symlinks, PATH modifications, Docker, VMs, Python virtual environment, ...), so it is not only about adding version number to the name. This is not to say that GRASS cannot help this in some way, but the default behavior should be that you get a basic `grass` command in one way or another without any extra steps.

Scripts, CI, etc. should generally just require change of the installed version (Git reference, package version, Docker image, ...) and the rest should work out of the box (assuming no changes in the API), i.e., without the need to go over the files and modify all references to the executable or other things to update the version. This should happen without the need to parametrize the version number and the need to pass it through all the files.

**Pull request merged**
